### PR TITLE
chore(datastore): Add github workflow for EndToEnd tests

### PR DESCRIPTION
**Pull request by lawmicha**
*Issue #, if available:*

*Description of changes:*
This PR adds datastore integration tests in the github actions worklows.

- .github/composite_actions/run_xcodebuild_test/action.yml is shared across unit test and other workflows like integ_test_datastore.yml so a project path was needed since the integration tests are nested in the repo like `./AmplifyPlugins/DataStore/Tests/DataStoreHostApp`
- override gitignore to include the Package.resolved from the HostApp, otherwise the build times out trying to resolve `aws-sdk-swift`.
- Set iOS deployment target of the host app to iOS ~~14~~ 13, otherwise, i saw failures in selecting the destination using `platform=iOS Simulator,name=iPhone 13,OS=latest`

image
https://github.com/aws-amplify/amplify-ios/runs/7769368982?check_suite_focus=true

*Check points: (check or cross out if not relevant)*

- [ ] Added new tests to cover change, if needed
- [ ] Build succeeds with all target using Swift Package Manager
- [ ] All unit tests pass
- [ ] All integration tests pass
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] Documentation update for the change if required
- [ ] PR title conforms to conventional commit style
- [ ] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ameter — COMMENTED** [review]
Are tests failing with an iOS deployment target of 13.0?  If so, we need to figure out why because we intend to support iOS 13.0+

**Comment by lawmicha** [comment]
> Are tests failing with an iOS deployment target of 13.0? If so, we need to figure out why because we intend to support iOS 13.0+

the error was
```
xcodebuild: error: Unable to find a destination matching the provided destination specifier:
		{ platform:iOS Simulator, OS:latest, name:iPhone 13 }

	Available destinations for the "AWSDataStorePluginIntegrationTests" scheme:
		{ platform:macOS, arch:x86_64, variant:Mac Catalyst, id:420[30](https://github.com/aws-amplify/amplify-ios/runs/7768909192?check_suite_focus=true#step:5:32)18E-580F-C1B5-9525-B745CECA79EB }

	Ineligible destinations for the "AWSDataStorePluginIntegrationTests" scheme:
		{ platform:iOS, id:dvtdevice-DVTiPhonePlaceholder-iphoneos:placeholder, name:Any iOS Device }
		{ platform:iOS Simulator, id:dvtdevice-DVTiOSDeviceSimulatorPlaceholder-iphonesimulator:placeholder, name:Any iOS Simulator Device }
		{ platform:macOS, variant:Mac Catalyst, name:Any Mac }
Error: Process completed with exit code 70.
```
i believe this was due to the host app being set to deployment target iOS 15.5, not an issue with iOS 13

**Comment by lawmicha** [comment]
Latest test run
image
https://github.com/aws-amplify/amplify-ios/runs/7771727931?check_suite_focus=true

File: .github/workflows/integ_test_datastore.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+name: DataStore Integration Tests
+on:
+  push:
+    branches: [dev-preview-ds]
```
Comment (lawmicha) [inline .github/workflows/integ_test_datastore.yml:4]:
```suggestion
    branches: [dev-preview]
```

**Review by ameter — APPROVED** [review]

**Pull request merged**